File: 455_AssignCookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        r, i = 0, 0
        g.sort()
        s.sort()
        for si in s:
            if i == len(g):
                break
            else:
                if si >= g[i]:
                    r, i = r + 1, i + 1
        return r

File: 455.AssignCookies.java
import java.util.Arrays;

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int i = 0;
        for (int j = 0; i < g.length && j < s.length; j += 1) {
            if (g[i] <= s[j]) {
                i += 1;
            }
        }
        return i;
    }
}
